src/sagas/__tests__/backend.ts

```diff
@@ -0,0 +1,265 @@
+import { expectSaga } from 'redux-saga-test-plan';
+import { call } from 'redux-saga/effects';
+
+import * as actions from '../../actions';
+import * as actionTypes from '../../actions/actionTypes';
+import { WorkspaceLocation } from '../../actions/workspaces';
+import {
+  AssessmentStatuses,
+  IAssessment,
+  IQuestion
+} from '../../components/assessment/assessmentShape';
+import {
+  mockAssessmentOverviews,
+  mockAssessmentQuestions,
+  mockAssessments
+} from '../../mocks/assessmentAPI';
+import { Role } from '../../reducers/states';
+import { showSuccessMessage, showWarningMessage } from '../../utils/notification';
+import backendSaga, {
+  getAssessment,
+  getAssessmentOverviews,
+  getUser,
+  postAnswer,
+  postAssessment,
+  postAuth
+} from '../backend';
+
+// ----------------------------------------
+// Constants to use for testing
+
+const mockAssessment: IAssessment = mockAssessments[0];
+
+const mockMapAssessments = new Map<number, IAssessment>(mockAssessments.map(a => [a.id, a]));
+
+const mockAssessmentQuestion = mockAssessmentQuestions[0];
+
+const mockTokens = { accessToken: 'access', refreshToken: 'refresherOrb' };
+
+// mock states here starts as student
+const mockStates = {
+  session: {
+    accessToken: 'access',
+    assessmentOverviews: mockAssessmentOverviews,
+    assessments: mockMapAssessments,
+    refreshToken: 'refresherOrb',
+    role: Role.Student
+  },
+  workspaces: {
+    assessment: { currentAssessment: 0 }
+  }
+};
+
+const okResp = { ok: true };
+const errorResp = { ok: false };
+// ----------------------------------------
+
+describe('Test FETCH_AUTH Action', () => {
+  test('when tokens and user obtained', () => {
+    const luminousCode = 'luminousCode';
+    const user = 'user';
+    return expectSaga(backendSaga)
+      .put(actions.setTokens(mockTokens))
+      .put(actions.setUser(user))
+      .provide([[call(postAuth, luminousCode), mockTokens], [call(getUser, mockTokens), user]])
+      .dispatch({ type: actionTypes.FETCH_AUTH, payload: luminousCode })
+      .silentRun();
+  });
+  test('when tokens is null', () => {
```

Comment:
This test is actually doing nothing. You should assert that `setTokens` / `setUser` is not being called.

Refer to https://redux-saga-test-plan.jeremyfairbank.com/integration-testing/negated-assertions.html